modules/core/shared/src/main/scala/gem/TargetEnvironment.scala

```diff
@@ -10,51 +10,60 @@ import scala.collection.immutable.SortedSet
 /** Collection of targets associated with an observation. */
 sealed trait TargetEnvironment {
   def asterism: Option[Asterism]
+  def guideEnvironment: GuideEnvironment
   def userTargets: SortedSet[UserTarget]
   override final def toString =
     s"TargetEnvironment($asterism, $userTargets)"
 }
 
 object TargetEnvironment {
 
-  def fromAsterism(a: Asterism, u: SortedSet[UserTarget]): TargetEnvironment =
+  def fromAsterism(
+    a: Asterism,
+    g: Option[GuideEnvironment],
+    u: SortedSet[UserTarget]
+  ): TargetEnvironment =
     a match {
-      case a @ Asterism.Phoenix(_)            => Phoenix(Some(a), u)
-      case a @ Asterism.Michelle(_)           => Michelle(Some(a), u)
+      case a @ Asterism.AcqCam(_)             => AcqCam(Some(a), u)
+      case a @ Asterism.Bhros(_)              => Bhros(Some(a), u)
+      case a @ Asterism.Flamingos2(_)         => Flamingos2(Some(a), GuideEnvironment.Flamingos2.typed(g), u)
```

Comment:
Here the `Option[GuideEnvironment]` we get could be anything but we can only assemble a valid `TargetEnvironment` if it is either `None` or a `Some(GuideEnvironment.Flamingos2(opts))`.